src/pipelines/stable_diffusion.rs

```diff
@@ -150,7 +154,11 @@ impl StableDiffusionConfig {
         let mut vs_unet = nn::VarStore::new(device);
         let unet =
             unet_2d::UNet2DConditionModel::new(vs_unet.root(), in_channels, 4, self.unet.clone());
-        vs_unet.load(unet_weights)?;
+        if unet_weights.ends_with("safetensors") {
```

Comment:
I suppose it's better to check the extensions with a dot:

```suggestion
        if unet_weights.ends_with(".safetensors") {
```